### PR TITLE
cohort 4: advanced-beginner pairing guidelines

### LINKED ISSUE #408: Add advanced-to-beginner pairing guidelines to playbook

### PR DESCRIPTION
**Pull request by maxwofford**
@hackedu/hack-camp Would love to get a review from one of our club leaders as well as someone who has paired successfully with someone of a vastly different skill level.

For the time being, assigning the PR to @zachlatta 

This closes #408

### CODE REVIEW

File: cohort_4/playbook/ACTIVITIES.md [hunk 1]
```diff
@@ -689,6 +689,35 @@ Uniqueness is also important - no two people should be paired together again
...
+## Types of Pairing
+
+### Pair programming between beginner-advanced pairs
+
+Pair programming between beginner and advanced hackers is fairly similar to
+normal pair programming. The most benefit will come from the beginner driving
+and the advanced hacker navigating, so rotating is not as important a factor in
+these situations (hackers should rotate if their bordom becomes a blocking
+factor for them to continue working together, but optimize for the beginner
+driving and the advanced hacker navigating as much as possible).
```
Comment (zachlatta) [inline cohort_4/playbook/ACTIVITIES.md]: They should still be rotating regularly. I believe the problem is that advanced programmers have a tendency to want to drive the coding completely, and we need to get them to resist that urge.

File: cohort_4/playbook/ACTIVITIES.md [hunk 2]
```diff
@@ -689,6 +689,35 @@ Uniqueness is also important - no two people should be paired together again
...
+
+When pairing between beginner and advanced hackers, make sure each hacker
+understands what they are getting out of it:
+
+For the advanced hacker:
+
+- True mastery of a subject can only come once someone is at the point they can
+  explain it to someone. If a hacker can't navigate while pairing, they dont
+  have an indepth understanding of the topic.
```
Comment (zachlatta) [inline cohort_4/playbook/ACTIVITIES.md]: `indepth` -> `in-depth`

File: cohort_4/playbook/ACTIVITIES.md [hunk 3]
```diff
@@ -689,6 +689,35 @@ Uniqueness is also important - no two people should be paired together again
...
+- This is practice in working with people who don't know as much
+  subject-specific information. Teaching this content to a beginning hacker is
+  practice for any future club leading activities.
```
Comment (zachlatta) [inline cohort_4/playbook/ACTIVITIES.md]: I'd suggest removing the "doing it in a club" context entirely – not all advanced coders will want to lead a club.

**Comment by zachlatta** [comment]
Left a couple comments and assigned back to you.

On Sun, Aug 9, 2015 at 4:56 AM, Max Wofford notifications@github.com
wrote:

> Assigned #411 https://github.com/hackedu/hack-camp/pull/411 to
> @zachlatta https://github.com/zachlatta.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/hackedu/hack-camp/pull/411#event-377342076.

## 

Zach Latta
hackEDU | zach@hackedu.us

**Comment by maxwofford** [comment]
@zachlatta Addressed all the things

File: cohort_4/playbook/ACTIVITIES.md [hunk 4]
```diff
@@ -689,6 +689,35 @@ Uniqueness is also important - no two people should be paired together again
 unless everyone has already been paired with each other. This will help members
 of the club all form bonds with each other.
 
+## Types of Pairing
+
+### Pair programming between beginner-advanced pairs
+
+Pair programming between beginner and advanced hackers is fairly similar to
+normal pair programming. They should still be rotating regularly. The advanced
+hacker will have a tendnacy to drive the coding completely and club leader may
```
Comment (zachlatta) [inline cohort_4/playbook/ACTIVITIES.md]: tendnacy -> tendency

File: cohort_4/playbook/ACTIVITIES.md [hunk 5]
```diff
@@ -689,6 +689,35 @@ Uniqueness is also important - no two people should be paired together again
...
+have to help them resist that urge.
```
Comment (zachlatta) [inline cohort_4/playbook/ACTIVITIES.md]: Perhaps make this a bit less declarative?

Ex. "Do note: we've observed that advanced hackers often have the tendency to want to drive the coding completely. You may need to have to help them resist that urge by doing x, y, and z."

File: cohort_4/playbook/ACTIVITIES.md [hunk 6]
```diff
@@ -689,6 +689,35 @@ Uniqueness is also important - no two people should be paired together again
...
+  subject-specific information. Teaching this content to a beginning hacker is
+  practice for any future interaction with teammembers.
+- A team is only as strong as its weakest link. Helping teammembers gain more
+  experience and increase their knowledge can pay off later.
+
+For the beginning hacker:
+
+- Focusing only on the code is a good way to get familiar with syntax and build
+  up muscle memory.
+- Anytime the hacker is lost, they can reach out to the navigator.
```
Comment (zachlatta) [inline cohort_4/playbook/ACTIVITIES.md]: Does this advice still apply if they're rotating regularly?

Comment (maxwofford) [inline cohort_4/playbook/ACTIVITIES.md]: I think it does. It doesn't matter who is at the computer, they should be able to ask each-other questions.

**Comment by maxwofford** [comment]
@zachlatta changed up the wording.

**Comment by maxwofford** [comment]
@zachlatta Ping

**Comment by maxwofford** [comment]
@zachlatta Would you like me to squash these.

**Comment by zachlatta** [comment]
Apologies for the delay! All looks well – it'd be great if you could squash
these. Then I'll be ready to merge it.

On Mon, Aug 10, 2015 at 12:39 PM, Max Wofford notifications@github.com
wrote:

> @zachlatta https://github.com/zachlatta Would you like me to squash
> these.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/hackedu/hack-camp/pull/411#issuecomment-129578791.

## 

Zach Latta
hackEDU | zach@hackedu.us

**Comment by maxwofford** [comment]
@zachlatta Squashed

**Comment by maxwofford** [comment]
@zachlatta nudge on this.

**Comment by zachlatta** [comment]
Is there a section in the table of contents for this?

**Comment by maxwofford** [comment]
@zachlatta Thanks for noticing that! Added.

**Pull request merged**